source/core/index.ts

```diff
@@ -1302,7 +1305,7 @@ export default class Request extends Duplex implements RequestEvents<Request> {
 		const responseEventName = options.cache ? 'cacheableResponse' : 'response';
 
 		request.once(responseEventName, (response: IncomingMessageWithTimings) => {
-			this._onResponse(response);
+			void this._onResponse(response);
```

Comment:
Are you sure `_onResponse` could never reject? The code in it is quite complicated so I don't think we should assume so. Better to be safe.

Reply:
The only possibility is when the `response` object is not an `IncomingMessage`. In that case it will be unhandled rejection. The solution would be to

```js
try {
	await this._onResponse(response);
} catch (error) {
	this._beforeError(error);
}
```

or just add the `try/catch` global clause inside `_onResponse`.

So, what should I do?

Reply:
Wrapping the whole function body would make the code hard to read, instead I would make a wrapper function that try/catches `_beforeError` and use that one instead everywhere.

- `_beforeErrorBase` - no try/catch
- `_beforeError` - try catches `_beforeErrorBase` and is the one we actually use.

Reply:
`_beforeError` never throws :P